omnipaxos_ui/src/app.rs

```diff
@@ -0,0 +1,76 @@
+use crate::util::defaults::*;
+use std::time::Instant;
+use omnipaxos::util::{ConfigurationId, NodeId};
+
+#[derive(Debug, Clone, Default)]
+pub struct Node {
+    pub pid: NodeId,
+    pub configuration_id: ConfigurationId,
+    pub ballot_number: u32,
+    pub connectivity: u8,
+}
+
+impl PartialEq for Node {
+    fn eq(&self, other: &Self) -> bool {
+        self.pid == other.pid
+            && self.configuration_id == other.configuration_id
+            && self.ballot_number == other.ballot_number
+    }
+}
+
+/// Ui application, containing the ui states
+pub struct App {
```

Comment:
We should hide the internal state as much as possible. I don't think the fields and `set_decided_idx` need to be public. The only things that should be visible for the users are the public functions in OmniPaxosUI that are needed to create, tick, and update the ui.